File: app/src/main/java/com/sevtinge/cemiuiler/module/hook/securitycenter/beauty/BeautyLightAuto.java
package com.sevtinge.cemiuiler.module.hook.securitycenter.beauty;

import com.sevtinge.cemiuiler.module.base.BaseHook;
import com.sevtinge.cemiuiler.module.hook.securitycenter.SecurityCenterDexKit;

import java.lang.reflect.Method;
import java.util.List;
import java.util.Objects;

import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedBridge;
import io.luckypray.dexkit.descriptor.member.DexMethodDescriptor;

public class BeautyLightAuto extends BaseHook {
    @Override
    public void init() {
        try {
            List<DexMethodDescriptor> result = Objects.requireNonNull(SecurityCenterDexKit.mSecurityCenterResultMap.get("BeautyLightAuto"));
            for (DexMethodDescriptor descriptor : result) {
                if (!String.valueOf(descriptor).contains("<clinit>")) {
                    Method beautyLightAuto = descriptor.getMethodInstance(lpparam.classLoader);
                    if (beautyLightAuto.getReturnType() == boolean.class && !String.valueOf(descriptor).contains(String.valueOf(BeautyFace.beautyFace))) {
                        log("beautyLightAuto method is " + beautyLightAuto);
                        XposedBridge.hookMethod(beautyLightAuto, XC_MethodReplacement.returnConstant(true));
                    }
                }
            }
        } catch (Throwable e) {
            e.printStackTrace();
        }
        /*
        int appVersionCode = getPackageVersionCode(lpparam);
        if (appVersionCode == 40000749) {
            findAndHookMethod("q7.m", "c", XC_MethodReplacement.returnConstant(true));
        } else {
            findAndHookMethod("com.miui.gamebooster.beauty.i", "i", XC_MethodReplacement.returnConstant(true));
        }
            /*findAndHookMethod("com.miui.gamebooster.utils.i", "i", new BaseHook.MethodHook() {
                @Override
                protected void before(XC_MethodHook.MethodHookParam param) throws Throwable {
                    param.setResult(true);
                }
            });
       */

    }
}

File: app/src/main/java/com/sevtinge/hyperceiler/module/hook/securitycenter/beauty/BeautyLightAuto.kt
/*
  * This file is part of HyperCeiler.

  * HyperCeiler is free software: you can redistribute it and/or modify
  * it under the terms of the GNU Affero General Public License as
  * published by the Free Software Foundation, either version 3 of the
  * License.

  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Affero General Public License for more details.

  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.

  * Copyright (C) 2023-2024 HyperCeiler Contributions
*/
package com.sevtinge.hyperceiler.module.hook.securitycenter.beauty

import com.github.kyuubiran.ezxhelper.*
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.sevtinge.hyperceiler.module.base.*
import com.sevtinge.hyperceiler.module.base.dexkit.*
import com.sevtinge.hyperceiler.module.base.dexkit.DexKitTool.addUsingStringsEquals
import com.sevtinge.hyperceiler.module.base.dexkit.DexKitTool.toElementList
import com.sevtinge.hyperceiler.module.base.dexkit.DexKitTool.toMethod
import de.robv.android.xposed.*

object BeautyLightAuto : BaseHook() {
    private val beauty by lazy {
        DexKit.getDexKitBridge("superWirelessCharge") {
            it.findMethod {
                matcher {
                    addUsingStringsEquals("taoyao")
                    returnType = "boolean"
                }
            }.single().getMethodInstance(EzXHelper.classLoader)
        }.toMethod()
    }
    private val beautyAuto by lazy {
        DexKit.getDexKitBridgeList("superWirelessCharge") {
            it.findMethod {
                matcher {
                    addUsingStringsEquals("taoyao")
                    returnType = "boolean"
                }
            }.toElementList(EzXHelper.classLoader)
        }.toMethodList()
    }

    override fun init() {
        if (mPrefsMap.getBoolean("security_center_beauty_face")) {
            beauty.createHook {
                returnConstant(true)
            }
        }

        beautyAuto.forEach {
            if (!java.lang.String.valueOf(it).contains("<clinit>")) {
                if (!java.lang.String.valueOf(it.name).contains(beauty.toString()) && it.name != beauty.name) {
                    logI(TAG, this.lpparam.packageName, "beautyLightAuto method is $beautyAuto")
                    XposedBridge.hookMethod(it, XC_MethodReplacement.returnConstant(true))
                }
            }
        }
    }
}
